src/apimtemplate/Core/Mapping/BackendMapper.cs

```diff
@@ -8,7 +9,12 @@ public class BackendMapper: IMapper
 	{
 		public void Map(IMapperConfigurationExpression cfg)
 		{
-			cfg.CreateMap<BackendDeploymentDefinition, BackendProperties>();
+			cfg.CreateMap<BackendDeploymentDefinition, BackendProperties>()
+				.ForMember(dst => dst.Properties, opt => opt.MapFrom(src => src.Properties))
+				.ForPath(dst => dst.Properties.ServiceFabricCluster.ClientCertificateId, 
+						opt => opt.MapFrom(src => string.IsNullOrEmpty(src.Properties.ServiceFabricCluster.ClientCertificateId) 
```

Comment:
should we do `src.?Properties?.ServiceFabricCluster?.ClientCertificateId`?

Reply:
It seems that Automapper automatically handles the null propagation. Also see [answer from StackOverflow](https://stackoverflow.com/a/29234649/8519078). I tested removing some & all backend properties and it didn't throw any exceptions.

Reply:
Actually I think we still need to check for null, I tested with [this yml](https://github.com/mirsaeedi/dotnet-apim/blob/master/src/Examples/Definitions/Backend.yml) and it's adding `serviceFabriccluster` property 
![image](https://user-images.githubusercontent.com/16405269/178312608-7ca93a11-3ad1-46e0-9348-69928af1f097.png)
![image](https://user-images.githubusercontent.com/16405269/178312671-dbfd1b30-4c3e-41bc-b101-d78f2eb89ee7.png)

I'll see how I can fix this.

Reply:
@mirsaeedi I'm stuck here. I tried this, but it still adds `serviceFabricCluster` even if it's not specified in the yml.
```
cfg.CreateMap<BackendDeploymentDefinition, BackendProperties>()
				.ForMember(dst => dst.Properties, opt => opt.MapFrom(src => src.Properties ?? null))
				.ForPath(dst => dst.Properties.ServiceFabricCluster, opt =>
				{
					opt.Condition(src => src.SourceMember != null);
					opt.MapFrom(src => src.Properties.ServiceFabricCluster ?? null);
				})
				.ForPath(dst => dst.Properties.ServiceFabricCluster.ClientCertificateId, opt =>
				{
					opt.Condition(src => src.Source?.Properties?.ServiceFabricCluster != null && !string.IsNullOrEmpty(src.SourceMember));
					opt.MapFrom(src => $"[resourceId('{ResourceType.Certificate}', parameters('ApimServiceName'), '{src.Properties.ServiceFabricCluster.ClientCertificateId}')]");
				});
```

Reply:
Okay I think I figured it out. Please review @mirsaeedi

Reply:
Hey @levimatheri, I commented 8 days ago but my comments were in the `Pending` status. Sorry for the confusion.